tree/ntuple/inc/ROOT/RNTupleTypes.hxx

```diff
@@ -99,23 +99,28 @@ enum class ENTupleColumnType {
    kMax,
 };
 
-/// The fields in the ntuple model tree can carry different structural information about the type system.
-/// Leaf fields contain just data, collection fields resolve to offset columns, record fields have no
-/// materialization on the primitive column layer.
+/// The fields in the RNTuple data model tree can carry different structural information about the type system.
+/// Collection fields have an offset column and subfields with arbitrary cardinality, record fields have no
+/// materialization on the primitive column layer and an arbitrary number of subfields. Static fields are either
+/// leafs (e.g., `float`) or fields with exactly one child that has the same cardinality as the parent
+/// (modulo field repetitions, e.g. std::atomic<T>).
```

Comment:
I don't 'grasp' the `static` semantic in this context.  To me `static field` evoke the idea describe here `https://en.cppreference.com/w/cpp/language/static.html`.
Is the intent to refer to the fact that the cardinality of the element is fixed within the context of its parent/container?  eg. Fixed size arrays or single value?  Or is it something else.

`Leaf` was evoking something that could be not further decomposed, for example numerical types and array thereof ... but also technically 'streamer field'.

Can you expand more on the rational of the choice of spelling for `kStatic`?

Reply:
I think the proper semantic association would be with "static array" (as opposed to dynamic array): something with a size known a-priori

Reply:
If/since this is the case I think we do need to extend the name to include this notion of size.

Reply:
Ok, let's try with "plain field".